website/common/script/content/loginIncentives.js

```diff
@@ -145,13 +147,89 @@ module.exports = function getLoginIncentives (api) {
         user.items.food.Saddle += 1;
       },
     },
+    55: {
+      rewardKey: ['Pet_HatchingPotion_Purple'],
+      reward: [api.hatchingPotions.RoyalPurple],
+      assignReward: function assignReward (user) {
+        if (!user.items.hatchingPotions.RoyalPurple) user.items.hatchingPotions.RoyalPurple = 0;
```

Comment:
We could in theory remove the check from all of them past the first, on the assumption that once you've got at least one potion you'll have the 0 value already. But that may not be safe.